velox/type/TimestampConversion.cpp

```diff
@@ -105,6 +105,8 @@ constexpr int32_t kCumulativeYearDays[] = {
 
 namespace {
 
+enum class ParseMode { kStrict, kNonStrict, kStandardCast, kNonStandardCast };
```

Comment:
nit: would be great if you can also add comments here as to what format each mode supports.